catalog/plugin/catalog-plugin-query-monitor-plugin/src/main/java/ddf/catalog/querymonitor/ActiveSearch.java

```diff
@@ -0,0 +1,205 @@
+     */
+
+    public ActiveSearch(String cqlQuery, Source source, UUID uniqueID, String clientInfo) {
+        if (cqlQuery == null) {
+            cqlQuery = "unknownCQLQuery";
+            LOGGER.debug("CQLQuery for new ActiveSearch is null. Being set as 'unknownCQLQuery'.");
+        }
+        this.cqlQuery = cqlQuery;
+        if (source == null) {
+            LOGGER.debug("Source for new ActiveSearch is null!");
+        }
+        this.source = source;
+        startTime = new Date();
+        if (uniqueID == null) {
+            this.uniqueID = java.util.UUID.randomUUID();
+            LOGGER.warn("uniqueID for new ActiveSearch is null. Setting the ID as " + uniqueID);
+        } else {
+            this.uniqueID = uniqueID;
+        }
+        if (clientInfo == null) {
+            this.clientInfo = "unknownClient";
+        } else {
+            this.clientInfo = clientInfo;
+        }
+    }
+
+    private static String getCqlFromQueryRequest(QueryRequest request) {
+        String cqlQuery;
+        if (request == null) {
+            LOGGER.warn("QueryRequest given to ActiveSearch constructor was null!");
+            return "";
+        } else {
+            cqlQuery = request.toString();
+            if (cqlQuery.equals("unknownCQLQuery")) {
+                LOGGER.warn("CQL Query not set as expected in QueryRequest");
+            } else {
+
+                String tempString = StringUtils.substringBetween(request.toString(),
+                        "filter=[",
+                        " ],startIndex");
+
+                if (tempString == null) {
+                    LOGGER.warn(
+                            "Unable to determine start and end of query terms. Using entire queryRequest String instead.");
+                } else {
+                    cqlQuery = tempString;
+                }
+            }
+        }
+        return cqlQuery;
+    }
+
+    public String toFormattedString() {
+        if (source == null) {
+            return String.format(FORMAT_STRING,
+                    clientInfo,
+                    "Unknown Source",
+                    cqlQuery,
+                    startTime.toString(),
+                    uniqueID.toString());
+        } else {
+            return String.format(FORMAT_STRING,
+                    clientInfo,
+                    source,
+                    cqlQuery,
+                    startTime.toString(),
+                    uniqueID.toString());
+        }
+    }
+
+    public ActiveSearch(Source source, QueryRequest request) {
+        this(getCqlFromQueryRequest(request), source, UUID.randomUUID(), "client");
+        if (request == null) {
+            LOGGER.warn("QueryRequest in ActiveSearch Constructor was null!");
+        } else {
+            clientInfo = SubjectUtils.getName((Subject) request.getPropertyValue(
+                    "ddf.security.subject"));
+            clientInfo = StringUtils.substringBetween(clientInfo, ",CN=", ",OU=");
```

Comment:
You should not be doing this, just use `SubjectUtils.getName(subject, "unknown", "true)` and then you can also remove the block of code under this line.